fs/sync/sync.go

```diff
@@ -152,6 +152,8 @@ func newSyncCopyMove(ctx context.Context, fdst, fsrc fs.Fs, deleteMode fs.Delete
 	} else {
 		s.ctx, s.cancel = context.WithCancel(ctx)
 	}
+	// Record source name in context for aware backends (see list() in Google Drive)
+	fi.SrcFsName = fsrc.Name()
```

Comment:
What you are trying to do is make the filters apply to just the backend in question.

However I don't like this approach - there could easily be multiple syncs running at once using the same `fi` block.

Perhaps what we need is an explicit marker to say to all backends - use the filters. This could be passed in the context.

This needs to pass down the value of the `includeAll` flag as used in `fs/walk` and `fs/march` - applying the flag to the context in those places will make it work for all backends.

So if we made a context variable `UseFilters` and applied it in those places in `fs/walk` and `fs/march` where we actually call `Fs.List` or `Fs.ListR` then this would fix it up.

Reply:
@ncw 
Done. PTAL!

Reply:
@ncw  Done! PTAL...